sbor/src/schema/schema.rs

```diff
@@ -51,6 +51,63 @@ impl<E: CustomTypeExtension> Schema<E> {
         }
     }
 
+    pub fn resolve_matching_tuple_metadata(
```

Comment:
So I'd repeated this kind of logic about 5 times - and thought it deserved a nicer abstraction - particularly the enum variant naming which is complicated.

I'm happy to add a rust doc to explain what they mean?

As for `tuple` - this is `tuple` as in the value kind :).

Reply:
Yes, would appreciate some documentations.

> As for tuple - this is tuple as in the value kind :).

If the schema matches payload; otherwise, we might be transmuting names.